Prevent blank screen

kryswisnaskas opened the pull request:
**Description of change**
This is the frontend and backend changes to prevent the application to result in a blank screen when the author is null. It also prevents unapproved legacy reports from showing up in the "My Alerts" table.

This PR also addresses changes for the Next Steps Screen Reader update and backend changes for the CSV activity report download.

**How to test**
- pull changes
- create an activity report in draft
- set it's author to null in the database (or in the MyAlerts.js file)
- verify that the UI works without showing a blank screen (when not on this branch, blank screen will show)

**Issue(s)**
* closes https://ocio-jira.acf.hhs.gov/browse/TTAHUB-91
* closes https://ocio-jira.acf.hhs.gov/browse/TTAHUB-57 
* closes https://ocio-jira.acf.hhs.gov/browse/TTAHUB-35 

**Checklist**
<!-- Add details to each completed item -->
- [x] Meets issue criteria
- [x] Code tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

jasalisbury reviewed (commented):
Should we link https://ocio-jira.acf.hhs.gov/browse/TTAHUB-57 and https://ocio-jira.acf.hhs.gov/browse/TTAHUB-35 to this PR?

dcmcand reviewed (commented):
This does through an error if the report is in submitted status, you remove the author, then click on the report (as the approving manager). Other than that it works great.

I could address that in https://github.com/adhocteam/Head-Start-TTADP/pull/269 though if you want

kryswisnaskas commented:
> I could address that in adhocteam#269 though if you want

Thanks! I updated the description.

> Should we link https://ocio-jira.acf.hhs.gov/browse/TTAHUB-57 and https://ocio-jira.acf.hhs.gov/browse/TTAHUB-35 to this PR?

Updated the description to include these two.
